### ISSUE TITLE
 DSHOT Telemetry in FW

### ISSUE THREAD

**Issue by BAI-djohnson** [PX4/PX4-Autopilot#20367]
I am attempting to receive DSHOT telemetry in my custom fixed wing for RPM and temperature. My system works if I select a MC so I know the wiring is correct but it is not working in FW airframe.

- PX4 1.13.0 Fixed wing airframe
- Cube Blue, standard carrier

DSHOT 1200 works correctly and I am able to send commands, but I am not receiving any telemetry back.  When I run dshot status, I get no successful frames and only timeouts.  Exact same wiring setup works with airframe selected as MC.

Any insight on how to get this working would be much appreciated.

**Comment by dagar** [comment]
I can't think of a reason it would be different between MC & FW.

Do you have logs or anything you can share between the working and non-working setup?

**Comment by BAI-djohnson** [comment]
I just remade the logs to make them smaller and do a true one to one comparison.  Please look at these instead.

Not working in FW with DSHOT telemetry: https://review.px4.io/plot_app?log=0ed87edf-3ed0-4c9e-b7d6-9fa163f33937
Working in MC with DSHOT telemetry: https://review.px4.io/plot_app?log=6a9c611d-662d-40ed-b988-3b791f07bca8

**Comment by BAI-djohnson** [comment]
![image](https://user-images.githubusercontent.com/95309010/194611694-31dd122b-120d-4081-a571-e66ebf2df906.png)
This appears to be where the issue is located.  
Located in dshot.cpp:255
I am not sure how to fix this without hard coding it.

**Comment by BAI-djohnson** [comment]
I also know this is not a correct solution but I was able to get logging of esc data by forcing the logger to include the topic by changing this to add_topic instead of add_optional_topic
![image](https://user-images.githubusercontent.com/95309010/194619429-399cfe59-3048-4e94-8400-b38fe49cbfad.png)

**Comment by bkueng** [comment]
@BAI-djohnson can you try with latest `main`? This will let you use the new control allocation interface, which should give you the right amount of motors. Docs for that are here: https://docs.px4.io/main/en/config/actuators.html.

**Comment by BAI-djohnson** [comment]
> @BAI-djohnson can you try with latest `main`? This will let you use the new control allocation interface, which should give you the right amount of motors. Docs for that are here: https://docs.px4.io/main/en/config/actuators.html.

Yes.  The second set of log files I uploaded were down with the current main and used the actuators tab for configuration.  Same issue, but I will attempt again today and report back if I have different results.
